### PR TITLE
Sample repository ontology

### PR DESCRIPTION
**Pull request by zhongh**

### CODE REVIEW

File: dco.ttl [hunk 1]
```diff
@@ -790,84 +790,66 @@ dco:vesselName
...
-                owl:allValuesFrom rdfs:Literal ;
-                owl:onProperty dco:hasElevation
+                owl:allValuesFrom dco:ResearchCommunity ;
+                owl:onProperty dco:associatedDCOCommunity
               ] ;
       rdfs:subClassOf
               [ a       owl:Restriction ;
-                owl:allValuesFrom rdfs:Literal ;
-                owl:onProperty dco:hasMaterial
+                owl:allValuesFrom core:GeographicLocation ;
```
Comment (mrpatrickwest) [inline dco.ttl]: This doesn't seem quite right. Might also be vivo:Address or other geographic representations. Just not sure how to represent it here.

**Comment by mrpatrickwest** [comment]
Example sample: https://udco.tw.rpi.edu/vivo/display/n11018
Example repository: https://udco.tw.rpi.edu/vivo/display/n11180

**Comment by mrpatrickwest** [comment]
Problem with pulling in the sampling ontology is that it imports quite a few different ontologies

https://udco.tw.rpi.edu/vivo/listOntologies

All the ones on this page that do not have a namespace label.

Should we add labels and annotations so they show up looking good?

**Comment by mrpatrickwest** [comment]
https://udco.tw.rpi.edu/vivo/display/n10039 - one of the repositories entered into UDCO ... not that bad. Please review.

**Comment by mrpatrickwest** [comment]
@zednis @olyerickson @zhonghao1014 @am-e @xgmachina 

please review

**Comment by mrpatrickwest** [comment]
New example sample: https://udco.tw.rpi.edu/vivo/display/n7813
New example repository: https://udco.tw.rpi.edu/vivo/display/n22533

File: dco-sample.ttl [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      rdfs:label "includes sample group"@en-US ;
+      rdfs:range dcosample:SampleGroup ;
+      rdfs:comment  """
+          a sample repository can include collections of samples as well as individual samples
+      """^^rdf:XMLLiteral .
+
+dcosample:SampleGroup
+      a       owl:Class ;
+      rdfs:label "Sample Group"@en-US ;
+      rdfs:subClassOf dco:Object ;
```
Comment (zednis) [inline dco-sample.ttl:103]: Should this also be a subclass of [`samfl:SamplingCollection`?](http://def.seegrid.csiro.au/ontology/om/sam-lite#SamplingCollection)

File: dco-sample.ttl [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      a       owl:ObjectProperty ;
+      rdfs:domain dcosample:Sample ;
+      rdfs:label "has sample type"@en-US ;
+      rdfs:range dcosample:SampleType ;
+      rdfs:subPropertyOf core:hasSubjectArea ;
+      rdfs:comment  """
+          A sample classification or annotation
+      """^^rdf:XMLLiteral .
+
+dcosample:hasIGSNRegistration
```
Comment (zednis) [inline dco-sample.ttl:58]: Should there be an inverse property for this property?

The IGSN Registration page has no link to the registered sample/feature https://udco.tw.rpi.edu/vivo/display/n16187

File: dco-sample.ttl (hunk 2 above)
Comment (mrpatrickwest) [inline dco-sample.ttl:103]: yes it should be

File: dco-sample.ttl (hunk 3 above)
Comment (mrpatrickwest) [inline dco-sample.ttl:58]: yes there should be an inverse property.

**Comment by xgmachina** [comment]
How about mapping a few properties to be sub-property of dct:hasPart and others to dct:isPartOf

**Comment by mrpatrickwest** [comment]
YOu mean instead of using hasPart and isPartOf directly?

**Comment by zednis** [comment]
Are we using hasPart and isPartOf directly?  I wasn't sure with `dcosample:includesSample` in the ontology.

`dcterms:isPartOf` instead of `dcosample:includesSample`?  Or make `dcosample:includesSample` a sub-property of `dcterms:isPartOf`?

**Comment by mrpatrickwest** [comment]
Not sure why I did it this way, but SampleGroup -> hasPart -> Sample -> isPartOf -> SampleGroup

**Comment by mrpatrickwest** [comment]
seems that should just be includesSample, remove the domain, add as restriction to SampleRepository and SampleGroup

**Comment by mrpatrickwest** [comment]
and make includesSample and includesSampleGroup subproperties of hasPart, and the inverse a subproperty of isPartOf.

**Comment by zednis** [comment]
> Not sure why I did it this way, but SampleGroup -> hasPart -> Sample -> isPartOf -> SampleGroup

This makes sense, though if SampleGroup is a `samfl:SamplingCollection` then it should use `samfl:member`.

```
samfl:SamplingCollection
  rdf:type owl:Class ;
  dct:description "Collection of features on which observations may be made, which are intended to be representative of one or more features that are not fully accessible for some reason. "^^xsd:string ;
  rdfs:comment "The member sampling features are intended to sample domain features, as indicated by one or more sampledFeature properties. If only one is present then this is the sampled feature of all the member sampling features, unless locally overridden."^^xsd:string ;
  rdfs:isDefinedBy <http://def.seegrid.csiro.au/ontology/om/sam-lite> ;
  rdfs:label "Collection of sampling features"@en ;
  rdfs:subClassOf [
      rdf:type owl:Restriction ;
      owl:minCardinality "1"^^xsd:nonNegativeInteger ;
      owl:onProperty samfl:member ;
    ] ;
.
```

**Comment by zednis** [comment]
> and make includesSample and includesSampleGroup subproperties of hasPart, and the inverse a subproperty of isPartOf.

:+1:

**Comment by mrpatrickwest** [comment]
okay ... to summarize

includesSample -> sub property of -> hasPart (inerse sub property of isPartOf)
includesSampleGroup -> sub property of -> hasPart (inerse sub property of isPartOf)

SampleGroup remove restrictions for hasPart since there's already a property samfl:member

Did I get that right?

everything is good.

**Comment by mrpatrickwest** [comment]
no inverse property for samfl:member so would add dcosample:memberOf to be the inverse property

**Comment by zhonghao1014** [comment]
👍

**Comment by zednis** [comment]
> includesSample -> sub property of -> hasPart (inerse sub property of isPartOf)
> includesSampleGroup -> sub property of -> hasPart (inerse sub property of isPartOf)

I think this is good; just to sanity check - this is the definition for `dcterms:hasPart`

> A related resource that is included either physically or logically in the described resource.

Do we all agree this makes sense as a super-property of includesSample and includesSampleGroup?
- does it make sense to say a sample is included physically or logically in the repository?
- does it make sense to say a sample group is included physically or logically in the repository?

**Comment by mrpatrickwest** [comment]
yes, I think that makes sense.

**Comment by xgmachina** [comment]
yes, too on the use of dct:hasPart

**Comment by am-e** [comment]
I like the use of dct:hasPart and dct:isPartOf, does that take care of our previous discussion about the relationship between samples/sample groups and the repository?

**Pull request merged**